server/models/video/sql/video/videos-id-list-query-builder.ts

```diff
@@ -260,6 +262,10 @@ export class VideosIdListQueryBuilder extends AbstractRunQuery {
       this.whereDurationMax(options.durationMax)
     }
 
+    if (options.excludeAlreadyWatched) {
```

Comment:
Please throw an error if `options.user.id` is not defined (you can use `exists` function)